doc/build-helpers/fetchers.chapter.md

```diff
@@ -70,28 +126,32 @@ fetchurl {
 }
 ```
 
-When building the recipe, you'll get an error message that you can use to determine the correct hash:
+When building the package, use the error message to determine the correct hash:
 
 ```shell
 $ nix-build
-(some output removed for clarity)
+...
 error: hash mismatch in fixed-output derivation '/nix/store/7yynn53jpc93l76z9zdjj4xdxgynawcw-version.drv':
          specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
             got:    sha256-BZqI7r0MNP29yGH5+yW2tjU9OOpOCEvwWKrWCv5CQ0I=
 error: build of '/nix/store/bqdjcw5ij5ymfbm41dq230chk9hdhqff-version.drv' failed
 ```
+:::
 
-[]{#fetchers-hash-tip}
-:::{.tip}
-For most cases, it is recommended to use the method illustrated above to figure out the appropriate hash to use in a fetcher:
-set the `hash` attribute to an empty string, try to build the derivation, and then copy the hash given by the error message.
+## Obtaining hashes securely {#sec-pkgs-fetchers-secure-hashes}
 
-Remember to set the `hash` attribute to an empty string whenever you update fetcher parameters and expect the content to change!
-:::
+Let's say Man-in-the-Middle (MITM) sits close to your network.
+Then instead of fetching source you can fetch malware, and instead of the actual source hash, you get the hash of malware.
```

Comment:
```suggestion
It's always a good idea to avoid Man-in-the-Middle (MITM) attacks when downloading source contents.
Otherwise, you could unknowingly download malware instead of the intended source, and instead of the actual source hash, you'll end up using the hash of malware.
```